engine/table/src/main/java/io/deephaven/engine/table/impl/SourcePartitionedTable.java

```diff
@@ -219,22 +241,66 @@ private void processPendingLocations(final boolean notifyListeners) {
-            resultRows.insert(added);
-            if (notifyListeners) {
-                result.notifyListeners(new TableUpdateImpl(added,
-                        RowSetFactory.empty(), RowSetFactory.empty(), RowSetShiftData.EMPTY, ModifiedColumnSet.EMPTY));
-            } else {
-                added.close();
-            }
+            final RowSet added = sortAndAddLocations(readyLocationStates.stream()
+                    .map(PendingLocationState::release));
             readyLocationStates.clearFast();
+            return added;
+        }
+
+        private RowSet processRemovals(final TableLocationSubscriptionBuffer.LocationUpdate locationUpdate) {
+            final Set<ImmutableTableLocationKey> relevantRemovedLocations = locationUpdate.getPendingRemovedLocations()
+                    .stream()
+                    .map(TableLocation::getKey)
+                    .filter(locationKeyMatcher)
+                    .collect(Collectors.toSet());
+
+            if (relevantRemovedLocations.isEmpty()) {
+                return RowSetFactory.empty();
+            }
+
+            final RowSetBuilderSequential deleteBuilder = RowSetFactory.builderSequential();
+
+            // We don't have a map of location to row key, so we have to iterate them. If we decide this is too slow, we
+            // could add a TObjectIntMap
+            // as we process pending added locations and then we can just make an index of rows to remove by looking up
+            // in that map.
+            try (final ChunkSource.GetContext locContext = resultTableLocationKeys.makeGetContext(CHUNK_CAPACITY);
+                    final ChunkSource.GetContext tableContext = resultLocationTables.makeGetContext(CHUNK_CAPACITY);
+                    final RowSequence.Iterator it = resultRows.getRowSequenceIterator()) {
+
+                while (it.hasMore()) {
+                    final RowSequence subSeq = it.getNextRowSequenceWithLength(CHUNK_CAPACITY);
+                    final LongChunk<OrderedRowKeys> keyChunk = subSeq.asRowKeyChunk();
+                    final ObjectChunk<Table, ? extends Values> tableChunk =
+                            resultLocationTables.getChunk(tableContext, subSeq)
+                                    .asObjectChunk();
+                    final ObjectChunk<ImmutableTableLocationKey, ? extends Values> removedKeys = resultTableLocationKeys
+                            .getChunk(locContext,
+                                    subSeq)
+                            .asObjectChunk();
+
+                    for (int chunkPos = 0; chunkPos < keyChunk.size(); chunkPos++) {
+                        if (relevantRemovedLocations.contains(removedKeys.get(chunkPos))) {
+                            deleteBuilder.appendKey(keyChunk.get(chunkPos));
+
+                            final Table deletedTable = tableChunk.get(chunkPos);
+                            result.unmanage(deletedTable);
+                            deletedTable.close();
```

Comment:
1. This `unmanage` should happen after downstream tables have processed the removes. Use an `UpdateCommitter`.
2. I'm not sure what you're on about with a `close()`. It's not really something we do. If you're going to do it, at least do it after cycle close.

Crap. This actually raises doubts about the entire PR. A removed location cannot be accessed safely (only cached data can be read), but we're telling downstream ops about the removed table here and kind of hoping they don't look at any of its data, which is almost certainly what every useful operation will want to do.